clang/include/clang/Sema/Sema.h

```diff
@@ -13072,27 +13069,42 @@ void Sema::addIntelSYCLSingleArgFunctionAttr(Decl *D,
     E = ICE.get();
     int32_t ArgInt = ArgVal.getSExtValue();
     if (CI.getParsedKind() == ParsedAttr::AT_SYCLIntelNumSimdWorkItems ||
-        CI.getParsedKind() == ParsedAttr::AT_IntelReqdSubGroupSize) {
+        CI.getParsedKind() == ParsedAttr::AT_IntelReqdSubGroupSize ||
+        CI.getParsedKind() == ParsedAttr::AT_IntelFPGAMaxReplicates) {
       if (ArgInt <= 0) {
         Diag(E->getExprLoc(), diag::err_attribute_requires_positive_integer)
-            << CI.getAttrName() << /*positive*/ 0;
+            << CI << /*positive*/ 0;
         return;
       }
     }
     if (CI.getParsedKind() == ParsedAttr::AT_SYCLIntelMaxGlobalWorkDim) {
       if (ArgInt < 0) {
         Diag(E->getExprLoc(), diag::err_attribute_requires_positive_integer)
-            << CI.getAttrName() << /*non-negative*/ 1;
+            << CI << /*non-negative*/ 1;
         return;
       }
       if (ArgInt > 3) {
         Diag(E->getBeginLoc(), diag::err_attribute_argument_out_of_range)
-            << CI.getAttrName() << 0 << 3 << E->getSourceRange();
+            << CI << 0 << 3 << E->getSourceRange();
+        return;
+      }
+    }
+    if (CI.getParsedKind() == ParsedAttr::AT_SYCLIntelSchedulerTargetFmaxMhz ||
+        CI.getParsedKind() == ParsedAttr::AT_IntelFPGAPrivateCopies) {
+      if (ArgInt < 0) {
+        Diag(E->getExprLoc(), diag::err_attribute_requires_positive_integer)
+            << CI << /*non-negative*/ 1;
         return;
       }
     }
   }
 
```

Comment:
Is there a reason you have added check (ArgInt < 0) again for SYCLIntelSchedulerTargetFmaxMhzAttr and IntelntelFPGAPrivateCopiesAttr here?

This is same as lines +13080 to + 13085. You can combine like
CI.getParsedKind() == ParsedAttr::AT_SYCLIntelMaxGlobalWorkDim ||
CI.getParsedKind() == ParsedAttr::AT_SYCLIntelSchedulerTargetFmaxMhz ||
CI.getParsedKind() == ParsedAttr::AT_IntelFPGAPrivateCopies

Reply:
SYCLIntelMaxGlobalWorkDim has an upper bound check on Line 13086, which is not required for SYCLIntelSchedulerTargetFmaxMhz  and IntelFPGAPrivateCopies. Instead of creating additional nested ifs, I kept the bound checks for SYCLIntelSchedulerTargetFmaxMhz and IntelFPGAPrivateCopies since the requirements are different. It felt cleaner this way.

Reply:
Yup, right.